src/compiler/program.ts

```diff
@@ -4302,15 +4302,38 @@ export function createProgram(rootNamesOrOptions: readonly string[] | CreateProg
         if (options.out) {
             createDeprecatedDiagnosticForOption(version, "out");
         }
+
+        forEachProjectReference(projectReferences, resolvedProjectReferences, (_resolvedRef, parent, index) => {
+            const ref = (parent ? parent.commandLine.projectReferences : projectReferences)![index];
+            if (ref.prepend) {
+                createDeprecatedOptionForVersionDiagnostic(
+                    version,
+                    (message, arg0, arg1, arg2) => createDiagnosticForReference(parent?.sourceFile as JsonSourceFile | undefined, index, message, arg0, arg1, arg2),
+                    "prepend",
+                );
+            }
+        });

```

Comment:
This loops over all projects and adds diagnostics to them, right? We don't seem to do that for any other deprecation warnings; is there something special about prepend that makes that necessary compared to the other deprecated compiler options?

Reply:
+1, it seems like this code belongs in the body of `verifyProjectReferences`

Reply:
I didnt add it in project references because didnt want to duplicate this check. But if you feel strongly i can do that. 

` if (ignoreDeprecations) {
            if (ignoreDeprecations === DeprecationVersion.v5_0 && (version === DeprecationVersion.v5_0 || version === DeprecationVersion.v5_5)) {
                return;
            }`
To reply to @jakebailey comment, we are not checking other options for deprecation because they are not relevant for the current project. The errors will be reported when building that project. But the "prepend" is in tsconfig that has reference to the project and the only place where the error can be issued.

Reply:
I do think it should just be in `verifyProjectReferences`; this function seems to only test compiler options (a different block in tsconfig), and it looks like `verifyProjectReferences` has an `if (ref.prepend) { ... }` place to issue diagnostics.

Reply:
That or, we need `verifyDeprecatedReferences` and move this code around a little?

Reply:
Hah, that would be equivalent to what you've written actually, since right now we have `verifyCompilerOptions` which calls `verifyDeprecatedCompilerOptions` and `verifyProjectReferences`. So, I think I'll take back my commentary on this seeming weird.

So, if done this way, `verifyCompilerOptions` calls `verifyDeprecatedCompilerOptions` and `verifyProjectReferences` calls `verifyDeprecatedProjectReferences` or something, containing exactly the code you've written...

Maybe that's the right answer; I don't know if we're going to be adding in more deprecations to the references block in the future.